pkg/cli/admin/release/extract.go

```diff
@@ -349,20 +354,42 @@ func (o *ExtractOptions) Run(ctx context.Context) error {
 		}
 	}
 
-	tarEntryCallbacks := []extract.TarEntryFunc{}
+	var versionInImageConfig string
+	opts.ImageConfigCallback = func(imageConfig *dockerv1client.DockerImageConfig) {
+		if imageConfig == nil {
+			klog.Info("Cannot retrieve the version because no image configuration is provided in the image to extract")
+			return
+		}
+		if c := imageConfig.Config; c != nil {
+			if v, ok := c.Labels["io.openshift.release"]; ok {
+				klog.V(2).Infof("Retrieved the version from image configuration in the image to extract: %s", v)
+				versionInImageConfig = v
+			} else {
+				klog.Info("Cannot retrieve the version from image configuration in the image to extract")
+			}
+		}
+	}
+
+	var manifestsCallbacks []func(string, []manifest.Manifest, io.Reader, []configv1.ClusterVersionCapability) (bool, error)
 
-	var manifestErrs []error
+	var needEnabledCapabilities bool
+	inclusionConfig := manifestInclusionConfiguration{}
 	if o.ExtractManifests {
 		expectedProviderSpecKind := credRequestCloudProviderSpecKindMapping[o.Cloud]
-
-		include := func(m *manifest.Manifest) error { return nil } // default to including everything
 		if o.Included {
 			context := "connected cluster"
-			inclusionConfig := manifestInclusionConfiguration{}
+			clientVersion, reportedVersion, err := version.ExtractVersion()
+			if err != nil {
+				return err
```

Comment:
This is complicated code, can we wrap this in a `fmt.Errorf("... %w")` and add some context about what the code is trying to do

Reply:
The relevant code is moved to https://github.com/openshift/oc/pull/2050

To be honest, I cannot really tell what `version.ExtractVersion()` is doing from its doc:

https://github.com/openshift/oc/blob/267009391e55919093984b63c22d67cb2de0c700/pkg/version/version.go#L79-L83

The code exists before this pull.
From the logging later, `reportedVersion` is used as the oc-cli's version.